packages/compass/src/app/components/home.tsx

```diff
@@ -152,13 +90,6 @@ function Home({
     [showCollectionSubMenu, hideCollectionSubMenu]
   );
 
-  useEffectOnChange(() => {
-    if (!isConnected) {
-      hideCollectionSubMenu();
```

Comment:
Same question here - is this not a thing in MC world?

Reply:
I don't think this meaningfully did anything for quite some time already (to be more specific I think it was only causing bugs in some corner cases where you'r "first connection" would change and just nuke the existing submenu unexpectedly), this is handled through the `onWorkspaceChanged` handler as a side-effect of active workspace changing for all cases